File: solutions/go/hamming/1/hamming.go
// Package hamming calculates the Hamming distance between two DNA strands
package hamming

import "errors"

// Distance calculates the Hamming distance between two strings
func Distance(a, b string) (int, error) {
	count := 0

	if len(a) != len(b) {
		return count, errors.New("DNA strands are not equal length")
	}

	for i := 0; i < len(a); i++ {
		if a[i] != b[i] {
			count++
		}
	}

	return count, nil

}

File: solutions/julia/hamming/1/hamming.jl
"""
    distance(a, b)

The Hamming distance between two DNA strands a and b.

"""
function distance(a, b)
    if length(a) != length(b)
        throw(ArgumentError("Strands not equal length."))
    end
    dist = 0
    for i in 1:length(a)
        if a[i] != b[i]
            dist += 1
        end
    end
    dist
end
